test/test_dynproc.py

```diff
@@ -164,14 +164,29 @@ def testJoin(self):
         size = MPI.COMM_WORLD.Get_size()
         rank = MPI.COMM_WORLD.Get_rank()
         server = client = address = None
-        # crate server/client sockets
+        host = socket.gethostname()
+        addresses = socket.getaddrinfo(host, None)
+        address_families = [ a[0] for a in addresses if a[1]==socket.SOCK_STREAM ]
+        # if both INET and INET6 are available, don't assume the order
+        # is the same on both server and client. Select INET if available.
+        if socket.AF_INET in address_families:
+            socket_family = socket.AF_INET
+        elif socket.AF_INET6 in address_families:
+            socket_family = socket.AF_INET6
+        elif socket.AF_UNIX in address_families:
+            socket_family = socket.AF_UNIX
+        elif address_families:
+            # allow for an undocumented family
+            socket_family = address_families[0]
+        else:
+            raise ConnectionError("internet socket not found")
```

Comment:
If networking is disabled, I do not want the test to fail, just skip it.
Please use this `raise unittest.SkipTest("socket")`.

Reply:
Will do.

Reply:
Just to check, I think that should be not `raise unittest.SkipTest("socket")` but rather
```
else:
    self.SkipTest("socket")
```
(or `self.SkipTest("network socket not found")`.  Do you prefer the shorter version?)

Reply:
Yes, you are definitely right, use the short version. 

BTW, in an upcoming PR, I should fix a few other cases of the longer version in favor of the shorter one.

Reply:
ok, I'll push self.skipTest("socket")